server/src/test/java/org/elasticsearch/discovery/AbstractDisruptionTestCase.java

```diff
@@ -52,17 +54,17 @@
 import java.util.HashSet;
 import java.util.List;
 import java.util.Set;
-import java.util.concurrent.ExecutionException;
 import java.util.concurrent.TimeUnit;
 
+import static org.elasticsearch.discovery.DiscoveryModule.DISCOVERY_HOSTS_PROVIDER_SETTING;
 import static org.hamcrest.Matchers.equalTo;
 import static org.hamcrest.Matchers.not;
 
 public abstract class AbstractDisruptionTestCase extends ESIntegTestCase {
 
     static final TimeValue DISRUPTION_HEALING_OVERHEAD = TimeValue.timeValueSeconds(40); // we use 30s as timeout in many places.
 
-    private ClusterDiscoveryConfiguration discoveryConfig;
+    private NodeConfigurationSource discoveryConfig;
 
     @Override
     protected Settings nodeSettings(int nodeOrdinal) {
```

Comment:
I would like to see if it's possible to pass in all the settings by implementing this method directly, rather than using the `NodeConfigurationSource` etc.

For instance it looks like these settings could be here:

```
            .put(FaultDetection.PING_TIMEOUT_SETTING.getKey(), "1s") // for hitting simulated network failures quickly
            .put(FaultDetection.PING_RETRIES_SETTING.getKey(), "1") // for hitting simulated network failures quickly
            .put("discovery.zen.join_timeout", "10s")  // still long to induce failures but to long so test won't time out
            .put(DiscoverySettings.PUBLISH_TIMEOUT_SETTING.getKey(), "1s") // <-- for hitting simulated network failures quickly
            .put(TransportService.TCP_CONNECT_TIMEOUT.getKey(), "10s") // Network delay disruption waits for the min between this
//and
                .put(TestZenDiscovery.USE_MOCK_PINGS.getKey(), false).build();
```

I also think these aren't needed and we can just make use of the machinery in `ESIntegTestCase` to set them appropriately:

```
                .put(NodeEnvironment.MAX_LOCAL_STORAGE_NODES_SETTING.getKey(), numberOfNodes)
                .put(ElectMasterService.DISCOVERY_ZEN_MINIMUM_MASTER_NODES_SETTING.getKey(), minimumMasterNode)
                .putList(DISCOVERY_HOSTS_PROVIDER_SETTING.getKey(), "file")
```

We use non-default settings in very few places; moving the affected tests into their own fixtures so they can override `nodeSettings` themselves seems worth investigating.

Reply:
@DaveCTurner Let me try :) What makes this a little tricky (requiring more changes maybe?) is that we have this code in `org.elasticsearch.discovery.ClusterDisruptionIT#testSearchWithRelocationAndSlowClusterStateProcessing`

```java
    /**
     * This test creates a scenario where a primary shard (0 replicas) relocates and is in POST_RECOVERY on the target
     * node but already deleted on the source node. Search request should still work.
     */
    public void testSearchWithRelocationAndSlowClusterStateProcessing() throws Exception {
        // don't use DEFAULT settings (which can cause node disconnects on a slow CI machine)
        configureCluster(Settings.EMPTY, 3, 1);
```

to override these defaults (though this could just be me not being so fluent in this code yet :)). Maybe do this in a follow up PR?

Reply:
Right, that's one of the tests that I think could be in its own fixture since it needs different cluster settings.

Reply:
@DaveCTurner wanna handle this here or can we move that to the next PR? (this one is already doing quite a few things)

Reply:
Sure, a follow-up is ok.